### DISCUSSION TITLE
How to write a retry logic script??

### DISCUSSION THREAD

**Discussion by RudraSen2** [actions/virtual-environments#5848] [General questions] [answered]
I want to write a retry logic script as mentioned here:

Unfortunately, nothing we can do from our side, add retry logic to a script.

_Originally posted by @al-cheb in https://github.com/actions/virtual-environments/issues/5824#issuecomment-1167252196_

How to do this?

**Comment by miketimofeev** [comment, marked as answer]
@RudraSen2 you can take a look at this example. We implement retry logic in download function. I believe you can wrap up your code into something similar https://github.com/actions/virtual-[environments/blob/054266a4b32735a980117226ec5357419366e777/images/macos/provision/utils/utils.sh#L3
](https://github.com/actions/virtual-environments/blob/054266a4b32735a980117226ec5357419366e777/images/macos/provision/utils/utils.sh#L19-L36)
